### PR TITLE
Ajax Practice, add widget to existing HTML file.

### PR DESCRIPTION
**Pull request by bfwetmore**
## Add widget to webpage for the following. 

- Add Feature to see Employees in/out status.
- Add feature to see Room status.

## Add files to support widget.js 

- JSON files to simulate Data to test.
- Add supporting CSS and HTML files.

### CODE REVIEW

File: ajax/widget/js/widget.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+let employeeList = new XMLHttpRequest();
+employeeList.onreadystatechange = function () {
```
Comment (do-work) [inline ajax/widget/js/widget.js]: This is an anonymous function.
I think you mostly want to use arrow functions nowadays.

File: ajax/widget/js/widget.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    if (employeeList.readyState === 4) {
```
Comment (do-work) [inline ajax/widget/js/widget.js]: I know you are following the courses but after you do the code try to review it for some obvious improvements. Looking at the code and getting a feel for how it flows will be invaluable over time.
So I would do a `if (employeeList.readyState !== 4) {return}`.
'Return first' reduces overall indentation which is always a bonus IF it does increase readability.

File: ajax/widget/js/widget.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        let employees = JSON.parse(employeeList.responseText);
+        let employeeStatus = '<ul class="bulleted">';
+        for (let i = 0; i < employees.length; i++) {
+            if (employees[i].inoffice === true) {
+                employeeStatus += '<li class="in">';
+            } else {
+                employeeStatus += '<li class="out">';
+            }
+            employeeStatus += employees[i].name;
+            employeeStatus += '</li>';
```
Comment (do-work) [inline ajax/widget/js/widget.js]: What about setting employeeStatus to `in` or `out` and then building employeeStatus entirely below the conditional?

```
           let employeeStatusClass = '';
           if (employees[i].inoffice === true) {
                employeeStatusClass = 'in';
            } else {
                employeeStatusClass = 'out';
            }
            employeeStatus += `<li class="${employeeStatusClass}">`
            employeeStatus += employees[i].name;
            employeeStatus += '</li>';
```

You could also do this where you set a default value up front and only change it if the conditional is met.
```
           let employeeStatusClass = 'out'
           if (employees[i].inoffice === true) {
                employeeStatusClass = 'in';
            }
            employeeStatus += `<li class="${employeeStatusClass}">`
            employeeStatus += employees[i].name;
            employeeStatus += '</li>';
```

**Review by do-work — CHANGES_REQUESTED** [review]

File: ajax/widget/js/widget.js (hunk 3 above)
Comment (bfwetmore) [reply ajax/widget/js/widget.js]: Done. I liked the default status idea most. Also inlined all 3 `employeeStatus` assignments into one template literal.

File: ajax/widget/js/widget.js (hunk 2 above)
Comment (bfwetmore) [reply ajax/widget/js/widget.js]: Done. Will try to keep an eye out for something like this.

File: ajax/widget/js/widget.js (hunk 1 above)
Comment (bfwetmore) [reply ajax/widget/js/widget.js]: I actually Just learned this yesterday! Done.

File: ajax/widget/js/widget.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    if (employeeList.readyState === 4) {return;}
+    let employees = JSON.parse(employeeList.responseText);
+    let employeeStatus = '<ul class="bulleted">';
+    for (let i = 0; i < employees.length; i++) {
+        let employeeStatusClass = "out";
+        if (employees[i].inoffice === true) {
+            employeeStatusClass = 'in';
+        }
+        employeeStatus += `<li class="${employeeStatusClass}">
+        ${employees[i].name}</li>`;
```
Comment (do-work) [inline ajax/widget/js/widget.js]: This looks weird  to me on two lines...
Can you put that on one line.

File: ajax/widget/js/widget.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+let employeeList = new XMLHttpRequest();
+
+employeeList.onreadystatechange = () => {
+    if (employeeList.readyState === 4) {return;}
```
Comment (do-work) [inline ajax/widget/js/widget.js]: I know I did the sample with this on one line, but I think most people prefer to see the return in its own block as in not inlined.
Also I had changed the conditional to a strict not equals.

File: ajax/widget/js/widget.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    employeeStatus += '</ul>';
+    document.getElementById('employeeList').innerHTML = employeeStatus;
+};
+
+employeeList.open('GET', 'data/employees.json');
+employeeList.send();
+
+let roomList = new XMLHttpRequest();
+roomList.onreadystatechange = () => {
+    if (roomList.readyState === 4) {return;}
```
Comment (do-work) [inline ajax/widget/js/widget.js]: Check this one too.

File: interacting-dom/app.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+const btnCreate = document.getElementById("btn-main");
+const btnToggle = document.querySelector(".btn-toggle");
+const btnRemove = document.querySelector(".btn-remove");
+const taskList = document.querySelector(".list-container ul");
+const remove = document.querySelectorAll('li');
```
Comment (do-work) [inline interacting-dom/app.js:5]: I would prefer more descriptive names than remove. It seems like that could be anything and isn't really descriptive of a list item..

File: interacting-dom/app.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    attachRemoveButton(remove[i]);
+}
+
+function attachRemoveButton(li) {
+    let remove = document.createElement('button');
+    remove.className = 'remove';
+    remove.textContent = 'Remove';
+    li.appendChild(remove);
+}
+taskList.addEventListener("click", (event) => {
```
Comment (do-work) [inline interacting-dom/app.js:16]: Should the target for this click event be more specific?
It looks like it is targeting the whole taskList and then checking if the button was selected.
Can you narrow it down to only attach the event listener to the button itself?

File: interacting-dom/app.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        btnToggle.textContent = "Hide List";
+        listContainer.removeAttribute("style");
+    } else {
+        btnToggle.textContent = "Show List";
+        listContainer.style.display = "none";
+    }
+});
+
+btnCreate.addEventListener("click", () => {
+    let ul = document.getElementsByTagName('ul')[0];
```
Comment (do-work) [inline interacting-dom/app.js:40]: Should you check the length before selecting the first element?

**Review by do-work — CHANGES_REQUESTED** [review]

File: ajax/widget/js/widget.js (hunk 6 above)
Comment (bfwetmore) [reply ajax/widget/js/widget.js]: Done

File: ajax/widget/js/widget.js (hunk 5 above)
Comment (bfwetmore) [reply ajax/widget/js/widget.js]: Done

File: ajax/widget/js/widget.js (hunk 4 above)
Comment (bfwetmore) [reply ajax/widget/js/widget.js]: Done.

**Pull request merged**